internal/common/retryattempt/retryattempt.go

```diff
@@ -0,0 +1,53 @@
+package retryattempt
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+)
+
+type Retryer struct {
+	delays []time.Duration
+}
+
+func NewRetryer(delays []time.Duration) *Retryer {
+	return &Retryer{delays: delays}
+}
+
+func NewOneAttemptRetryer() *Retryer {
+	return NewRetryer([]time.Duration{})
+}
+
+// Try Attempts retryable operation. Returns retry attempts count always and error on fail.
+func (r *Retryer) Try(ctx context.Context, do func() error, isRetryable func(err error) bool) (int, error) {
+	tmr := time.NewTimer(0)
+	tmr.Stop()
```

Comment:
Вопрос - таймер от нуля, который еще и остановлен будет ли вообще тикать? 
```suggestion
	defer tmr.Stop()
```
Не заметил вызов `Reset(...)`. Мне кажется тут трудно разобраться в том, как таймер используется. Мы создаем таймер от нуля, останавливаем его, делаем ресет, останавливаем. Можно как-то упростить? Попробуй посмотреть в сторону `time.After(...)`.

Reply:
++